apps/web/src/contexts/currency-context.tsx

```diff
@@ -0,0 +1,53 @@
+import { CURRENCY_COOKIE, DEFAULT_CURRENCY } from '@algomart/schemas'
+import { createContext, useContext, useState } from 'react'
+
+import { useAuth } from '@/contexts/auth-context'
+import { AuthService } from '@/services/auth-service'
+import { getCookie } from '@/utils/cookies-web'
+import { setCurrencyCookie } from '@/utils/cookies-web'
+
+interface ICurrencyContext {
+  currency: string
+  updateCurrency: (currency: string) => Promise<boolean>
+}
+
+const CurrencyContext = createContext<ICurrencyContext>({
+  currency: DEFAULT_CURRENCY,
+  updateCurrency: () => Promise.resolve(true),
+})
+
+export const useCurrency = () => useContext(CurrencyContext)
+
+export const CurrencyProvider = ({
+  children,
+}: {
+  children: React.ReactNode
+}) => {
+  const auth = useAuth(false)
+
+  // We need to store the cookie value in state in order to force re-render on change.
+  const [cookie, setCookie] = useState(getCookie(CURRENCY_COOKIE))
```

Comment:
Is this safe to be called here? Since it would be called both on the server and in the browser (`js-cookie` is a browser library).

Reply:
ooh I wonder if we can fix all our hydration issues by moving those checks here.